text/204-golang-bindings.md

```diff
@@ -0,0 +1,669 @@
+attribute that indicates that the interface only contains properties. While this does corresponds directly to a Go struct, we would likely need to
+generate both a Go interface that contains getter methods that correspond to each property as well as a Go struct that implements that interface. This
+is in order to support subtyping, as the interface is typically what is passed as an argument into other functions, as well as to ensure forward
+compatibility in case the datatype interface eventually extends another. Were it not for these considerations, it would be simpler to simply have a
+single Go struct that corresponds to a datatype interface (see last bullet point in Notes/Concerns).
+
+#### Case 1: Typescript datatype interface (no extensions)
+
+Here, a datatype interface (`HealthCheck`) is converted into a Go interface (`HealthCheck`), where each property becomes a
+getter method (JSII datatype properties always are readonly, so there are no setter methods generated). This Go interface is then implemented by a struct
+(LinuxParameterProps), which would contain the properties from the original typescript interface as non-exported (i.e. lowercase) properties and rely
+on the generated Getter methods for readonly access (as when _used as an argument_ to the constructor):
+
+[Example](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-ecs/lib/container-definition.ts#L596):
+
+```ts
+export interface HealthCheck {
+  readonly command: string[];
+  readonly interval?: cdk.Duration;
+  readonly retries?: number;
+  readonly startPeriod?: cdk.Duration;
+  readonly timeout?: cdk.Duration;
+}
+
+function renderHealthCheck(hc: HealthCheck): CfnTaskDefinition.HealthCheckProperty {
+  return {
+    command: getHealthCheckCommand(hc),
+    interval: hc.interval != null ? hc.interval.toSeconds() : 30,
+    retries: hc.retries !== undefined ? hc.retries : 3,
+    startPeriod: hc.startPeriod && hc.startPeriod.toSeconds(),
+    timeout: hc.timeout !== undefined ? hc.timeout.toSeconds() : 5,
+  };
+}
+```
+
+Go translation (Run example in the [Go playground](https://play.golang.org/p/_eRLhVmXmp8):
+
+```go
+package ecs
+
+type HealthCheckIface interface{
+   Command() []string
+   Interval() cdk.Duration;
+   Retries() int
+   StartPeriod() cdk.Duration;
+   Timeout() cdk.Duration;
+}
+
+type HealthCheck struct {
+    command []string
+    interval cdk.Duration;
+    retries int
+    startPeriod cdk.Duration;
+    timeout cdk.Duration;
+}
+
+func (h HealthCheck) Command() []string {
+    return h.command
+}
+
+func (h HealthCheck) Retries() int {
+    return h.retries
+}
+
+func (h HealthCheck) Interval() cdk.Duration {
+    return h.interval
+}
+
+func (h HealthCheck) StartPeriod() cdk.Duration {
+    return h.startPeriod
+}
+
+func (h HealthCheck) Timeout() cdk.Duration {
+    return h.timeout
+}
+
+func renderHealthCheck(hc HealthCheck) CfnTaskDefinition.HealthCheckProperty {
+    return CfnTaskDefintion.HealthCheckProperty{
```

Comment:
In this example would the package name be `CfnTaskDefinition`? Is the plan for CDK Go packages to use cammel casing?

Reply:
Ah, no. This would likely be `cfnTaskDefinition` or `cfntaskdefinition`, though the verbosity of the JSII FQNs make their conversion into idiomatic go package names a little clunky. This is mentioned in the [Packages and Submodles](https://github.com/SoManyHs/aws-cdk-rfcs/blob/master/text/204-golang-bindings.md#packages-and-submodules) section, though it's not consistent in the examples -- will fix!